### ISSUE TITLE
Touch/tap on any element - no immediate response

### ISSUE THREAD

**Issue by avalx** [mozilla-mobile/fenix#16235]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
Have to touch/tap on anything (link, button, menu, ...) a lot of times to get response from Firefox.
### Expected behavior
To have instant response for every touch/tap.
### Actual behavior
I have to touch/tap an element a lot of times (3,4,5,6,...) to get Firefox to react. Sometimes there is immediate response but mostly something happens only after a couple of taps - not always the same number. This is happening after I updated to version 82.1.1. If I downgrade to version 81.1.5 everything works as expected.
### Device information

* Android device: Oneplus 7T, OxygenOS 10.0.14 HD65BA, rooted with Magisk 21, Arter97 kernel (but the same problem is with stock OOS kernel)
* Fenix version: 82.1.1 (also tried 83.0.0-beta and have the same problem)

### Additional info
On my two other devices (Oneplus 3/Bliss ROM 12.12; Samsung Tab A/Custom Nexus ROM) this problem is not present.

**Comment by fraser-mac66** [comment]
I'm having the same issue, loss of touch sensitivity or misaligned touch elements, is the best way I can describe it. Multiple taps in multiple areas, close to the element, are required to open/ activate said element.

The problem is present on: 
- Firefox Daylight 82.1.1
- Firefox beta 83.0.0-beta.2 
- Firefox Nightly 201028. 

I don't think this was an issue on the previous version of FF Daylight, but I've had the same issue w/ Firefox beta and Firefox Nightly going back a couple of months now.

Pixel 3a XL
Android 11

**Comment by silvaran** [comment]
Made a support post before I found the right keywords to start turning up other results: https://support.mozilla.org/en-US/questions/1310674

**Comment by Mugurell** [comment]
Thank you for your reports!
Could you please share an example of a website you've seen this happening on? Would help us greatly.

**Comment by avalx** [comment]
> Made a support post before I found the right keywords to start turning up other results: https://support.mozilla.org/en-US/questions/1310674

Yes, I forgot to mention this problems also affects pinches as well.

**Comment by avalx** [comment]
> Thank you for your reports!
> Could you please share an example of a website you've seen this happening on? Would help us greatly.

This occurs on almost every site that I tried, but here are a couple:
github.com
f-droid.org
http://vreme.arso.gov.si/napoved/Ljubljana/graf (sorry, slovenian only)
https://www.openandromaps.org (hamburger menu on top right for example)
https://www.theregister.com

**Comment by avalx** [comment]
And the video (if higher resolution is needed, please let me know).


[20-10-29-08-21-15-1.zip](https://github.com/mozilla-mobile/fenix/files/5456816/20-10-29-08-21-15-1.zip)

**Comment by Mugurell** [comment]
> And the video (if higher resolution is needed, please let me know).
> 
> [20-10-29-08-21-15-1.zip](https://github.com/mozilla-mobile/fenix/files/5456816/20-10-29-08-21-15-1.zip)

Thank you. The issue is clearly demonstrated on f-droid.
Though I'm not able to reproduce it right away.
Also the hamburger menu from `openandroimaps.org` seems to always work.

Going by comment 0 I see this is not reproduceable on all devices but started happening with version 82.1.1 so maybe there is something we can do about it.
Does this happen in other browsers (other than Firefox) as well? 
Any other details that could help pinpoint the issue are welcome.

Asking QA to also check this to maybe find the app / device configuration on which this would be happening.

**Comment by avalx** [comment]
This

> > And the video (if higher resolution is needed, please let me know).
> > [20-10-29-08-21-15-1.zip](https://github.com/mozilla-mobile/fenix/files/5456816/20-10-29-08-21-15-1.zip)
> 

> Does this happen in other browsers (other than Firefox) as well?
> Any other details that could help pinpoint the issue are welcome.
> 
> Asking QA to also check this to maybe find the app / device configuration on which this would be happening.

**Comment by avalx** [comment]
Sorry I closed this issue by accident and have to reopen it! :(

**Comment by sflorean** [comment]
Checked on Firefox for Android 82.1.1 and Nightly 29/10 and I couldn't reproduce the issue with Pixel 3 (Android 11), Samsung Galaxy Note 10(Android 10), and OnePlus 6T (Android 9).

**Comment by alungoodger** [comment]
I have been seeing this issue a *lot* on a Oneplus 6T (Android 10)... this started (or got worse) with a recent update.
I have been seeing the issue on a private personal website which is not publicly available.It does not happen 100% consistently but the impression I am getting is that the touch processing in Firefox sometimes does not "wake up immediately", but once its up an running its for the most part ok. I find it very easy to reproduce by letting the phone screen turn off for a while, waking it up, selecting a home screen icon to launch the website and immediately attempting to use the touch screen to zoom in. Usually the first attempt to zoom in fails, but after that there is a higher probability of it working. Sometimes it takes 2-3 goes to get the zoom to work. For me once it is working its not such a noticeable problem, and tends to work until I let the screen turn off which tends to reset it so its broken on next launch.

**Comment by ThomasRadeke** [comment]
I have a LG G8s Thinq with Android 10 and I have the same issues. Unfortunately, the non-responding touch events also affect scrolling - I have to swipe the screen several times to register a scroll on _any_ website (not in menus). This is borderline unusable for me.

**Comment by avalx** [comment]
FWIW, two days ago the Tor browser was updated and now the same symptoms are present there as well.

![Screenshot_20201105-094352](https://user-images.githubusercontent.com/22810741/98243543-623d9a00-1f6e-11eb-8cc7-449a8140a520.jpg)

**Comment by Imerion** [comment]
l can confirm this issue on most pages using the latest beta as well as versions from a few weeks back. It especially affects smaller clickable areas, like text links. It might be due to some form of delay, because it often feels like it works more reliably as time has passed. I'm running Android 8.1 on a Blackberry KeyONE.
  I can confirm this issue on most pages, especially on smaller clickable areas like text links. It might be due to some form of delay, because it often feels

**Comment by silvaran** [comment]
I enabled touch input display through Developer Options on my phone and recorded 3 instances in less than 40 seconds of Firefox simply ignoring swipes: https://youtu.be/VktuFLnjhzU

It's just Firefox, and it's most--if not all websites, most of the time. I just tried to click the Comment button and Firefox ignored it so I went back and added this final sentence.

**Comment by meson11** [comment]
seems to be connected with this issue: #16235

**Comment by Tarkin77** [comment]
Hello,
i have the same issue and still with Firefox 82.1.3 also using a OnePlus 7T (Stock). Makes Firefox unusable :-(
On Samsung Galaxy S7 problem does not appear using same Firefox version.
Many people in Play Store comments are complaining about this issue giving Firefox lowest rating :-(

**Comment by LaurentiuApahideanSV** [comment]
I tried to reproduce the issue on Firefox Preview Nightly 201110 and 83.0.0-beta.4 but was unable to. If anyone has further information on how to reproduce the issue please let us know.
### Devices used:
- OnePlus 6T (Android 9)
- Pixel 3XL (Android 9)

**Comment by alungoodger** [comment]
> I tried to reproduce the issue on Firefox Preview Nightly 201110 and 83.0.0-beta.4 but was unable to. If anyone has further information on how to reproduce the issue please let us know.
> ### Devices used:
> 
>     * OnePlus 6T (Android 9)
> 
>     * Pixel 3XL (Android 9)

On my OnePlus 6T (Android 10) it happens all of the time, ie it's not limited to specific sites or any other factors

**Comment by ekager** [comment]
Can any of you please check if you can reproduce in Reference Browser? https://play.google.com/store/apps/details?id=org.mozilla.reference.browser

That would help us tell if it's an engine regression or on the Fenix side. Thank you!

**Comment by silvaran** [comment]
> Can any of you please check if you can reproduce in Reference Browser? https://play.google.com/store/apps/details?id=org.mozilla.reference.browser

I can't get that URL to load on my phone or on my desktop.  Desktop just reports "Not Found", and the phone just has a loading spinner that spins indefinitely and doesn't load anything.

**Comment by silvaran** [comment]
I had to join this group before I could see the reference browser in the Google Play store: https://groups.google.com/g/mozilla-reference-browser

I can confirm that the behaviour exists on the Reference Browser: https://www.youtube.com/watch?v=7kIxN434-s0

**Comment by avalx** [comment]
I can also confirm that the behaviour is as bad as with all other versions since 82.1.1.

Thanks to @silvaran for the tip about joining the group!

**Comment by ekager** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1676480

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).

**Comment by Tarkin77** [comment]
Problem has been fixed in Nightly 201118 :-)

**Comment by Mugurell** [comment]
Adding the `qa:needed` flag for verifying this and also the ones linked in this ticket.

**Comment by avalx** [comment]
I will try this as soon as this Nightly gets into Play Store (I guess tomorrow?).

**Comment by Tarkin77** [comment]
@avalx  You can get it here:

https://github.com/mozilla-mobile/fenix/wiki
-> https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.nightly.latest
-> arm64 https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.nightly.latest.arm64-v8a/artifacts/public/build/arm64-v8a/target.apk

**Comment by avalx** [comment]
So I can confirm that the problem is indeed gone with the latest nightly, at least for me. 😃

Thanks for your efforts and good work. 👍

**Comment by sflorean** [comment]
Tested on Nightly 11/19 and 11/18 following the steps provided here, and we couldn't reproduce the issue. 
@silvaran @alungoodger @ThomasRadeke could you please check if the issue is fixed for you, also? Thank you!
Devices:
- Pixel 3(Android 11)
- LG G7 (Android 8.0)
- Xiaomi Redmi Note 8T (Android 9)

**Comment by ThomasRadeke** [comment]
> Tested on Nightly 11/19 and 11/18 following the steps provided here, and we couldn't reproduce the issue.
> @silvaran @alungoodger @ThomasRadeke could you please check if the issue is fixed for you, also? Thank you!
Seems to be fixed for me, too, on a LG G8s ThinQ. Thank you!

**Comment by dhlocker** [comment]
Similar/same problem on 83.1.0 Firefox for Android ("Updated on 18 Nov 2020") no extensions, additions, ...) on Samsung Galaxy S5. Primarily problem with Facebook (m.facebook.com)

Most noticable when touching the "like" button; often will not react (normal operation is to raise the like/love/care/haha/wow/sad/angry icons). This morning, I could not get a touch on the "add to collection" above a post after successfully "save post." I could get the "like" from the previous post or the "checkmark" (indicating a verified site) for the current post but the very large "add to a collection" button between those two posts was not recognised. Screenshot attached.

![Screenshot_20201202-042410_Firefox](https://user-images.githubusercontent.com/5805239/100903116-86b86380-3493-11eb-9f7a-c6dccb1c09e5.png)

**Comment by Tarkin77** [comment]
@dhlocker 
The problem ist not fixed yet in the "normal" Firefox Release. It has been fixed in Firefox Nightlies and Beta since Beta 84.0.0-beta.2
